Sources/libsocket/socket-posix.c

```diff
@@ -112,7 +112,7 @@ socket_t socket_connect(const char* address, uint16_t port) {
     freeaddrinfo(result);
 
     if (addr == NULL) {
-       return -2;
+       return -1;
```

Comment:
Ah yes, I forget that these functions can return other values as well. Good catch! How about we move the `INVALID_SOCKET` definition into socket.h and turn all of these `-1` values into the `INVALID_SOCKET` constant? I think that'd be a bit nicer if it works.
Also, I think `socket_accept` can still return invalid values now.

Reply:
Sunk the definition into `libsocket` and updated the implementation.